packages/aws-cdk-lib/aws-cognito/lib/user-pool-client.ts

```diff
@@ -391,11 +392,11 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
     };
 
     let callbackUrls: string[] | undefined = props.oAuth?.callbackUrls;
-    if (this.oAuthFlows.authorizationCodeGrant || this.oAuthFlows.implicitCodeGrant) {
-      if (callbackUrls === undefined) {
-        callbackUrls = ['https://example.com'];
-      } else if (callbackUrls.length === 0) {
-        throw new Error('callbackUrl must not be empty when codeGrant or implicitGrant OAuth flows are enabled.');
+    if (!props.oAuth && callbackUrls === undefined) {
+      callbackUrls = ['https://example.com'];
+    } else if (props.oAuth?.flows?.authorizationCodeGrant || props.oAuth?.flows?.implicitCodeGrant) {
+      if (callbackUrls === undefined || callbackUrls.length === 0) {
+        throw new Error('callbackUrls must not be empty when codeGrant or implicitGrant OAuth flows are enabled.');
```

Comment:
I don't think that gives the same result. 

```
let callbackUrls: string[] | undefined = props.oAuth?.flows?.clientCredentials
      ? ['https://example.com']
      : props.oAuth?.callbackUrls;
```

This will set callbacks to `example.com` as long as `clientCredentials` is true. We don't want that. We want to use the callbacks passed in and fallback to `example.com` if needed.

Reply:
Ah true, you're right.
So a callback URL always must be set, right?
When looking at the UI, it says this:
![5343](https://github.com/aws/aws-cdk/assets/54506108/41c6b1c4-5142-4ddd-84f9-c290b62a9f24)

It seems like there is no distinction between each oAuth flow.